### PR TITLE
fixed bug on nuopc cap

### PR DESCRIPTION
**Pull request by mvertens**
### Description of changes
Changes in nuopc cap to fix bugs found in testing

### Specific notes

Contributors other than yourself, if any:

CTSM Issues Fixed (include github issue #):

Are answers expected to change (and if so in what way)?

Any User Interface Changes (namelist or namelist defaults changes)?

Testing performed, if any:
(List what testing you did to show your changes worked as expected)
(This can be manual testing or running of the different test suites)
(Documentation on system testing is here: https://github.com/ESCOMP/ctsm/wiki/System-Testing-Guide)
(aux_clm on cheyenne for intel/gnu and izumi for intel/gnu/nag/pgi is the standard for tags on master)

**NOTE: Be sure to check your Coding style against the standard:**
https://github.com/ESCOMP/ctsm/wiki/CTSM-coding-guidelines

### CODE REVIEW

File: src/cpl/nuopc/lnd_import_export.F90 [hunk 1]
```diff
@@ -197,8 +197,8 @@ subroutine advertise_fields(gcomp, flds_scalar_name, glc_present, rof_prognostic
 
     ! Fire emissions fluxes from land
     call shr_fire_emis_readnl('drv_flds_in', emis_nflds)
-    if (shr_fire_emis_mechcomps_n .ne. emis_nflds) call shr_sys_abort('ERROR: fire_emis field count mismatch')
-    if (shr_fire_emis_mechcomps_n > 0) then
+    write(6,*)'DEBUG: emis_nflds= ',emis_nflds
```
Comment (billsacks) [inline src/cpl/nuopc/lnd_import_export.F90]: This write to unit 6 is suspicious. Should this be writing to iulog?

Comment (mvertens) [reply src/cpl/nuopc/lnd_import_export.F90]: No - this line should be removed. thanks for catching this. I'll fix this and push it back.

**Comment by billsacks** [comment]
I'm running into trouble testing the nuopc build.

The first problem I ran into was an error at create_newcase time, `Makes no sense to have empty read-only file`. I realized that this was because the nuopc external wasn't being checked out. I fixed this with the following change in CTSM's Externals.cfg; should we put this in place moving forward?:

```diff
diff --git a/Externals.cfg b/Externals.cfg
index 82fa0b5f..f95c55f0 100644
--- a/Externals.cfg
+++ b/Externals.cfg
@@ -31,6 +31,7 @@ local_path = cime
 protocol = git
 repo_url = https://github.com/ESMCI/cime
 tag = branch_tags/cime5.8.3_chint17-05
+externals = ../Externals_cime.cfg
 required = True
 
 [externals_description]
```

With that in place, I got an error in the driver build:

```
/glade/work/sacks/ctsm_code/current_branch1/cime/src/drivers/nuopc/mediator/med_phases_aofluxes_mod.F90(500): error #6784: The number of actual arguments cannot be greater than the number of dummy arguments.   [SHR_FLUX_ATMOCN]
    call shr_flux_atmocn (&
---------^
/glade/work/sacks/ctsm_code/current_branch1/cime/src/drivers/nuopc/mediator/med_phases_aofluxes_mod.F90(508): error #6633: The type of the actual argument differs from the type of the dummy argument.   [0]
         0, aoflux%duu10n, ustar_sv=aoflux%ustar, re_sv=aoflux%re, ssq_sv=aoflux%ssq, &
---------^
/glade/work/sacks/ctsm_code/current_branch1/cime/src/drivers/nuopc/mediator/med_phases_aofluxes_mod.F90(500): error #6630: By using keywords, an actual argument has been associated more than once with the same dummy argument.   [USTAR_SV]
    call shr_flux_atmocn (&
---------^
/glade/work/sacks/ctsm_code/current_branch1/cime/src/drivers/nuopc/mediator/med_phases_aofluxes_mod.F90(508): error #6638: An actual argument is an expression or constant; this is not valid since the associated dummy argument has the explicit INTENT(OUT) or INTENT(INOUT) attribute.   [0]
         0, aoflux%duu10n, ustar_sv=aoflux%ustar, re_sv=aoflux%re, ssq_sv=aoflux%ssq, &
---------^
/glade/work/sacks/ctsm_code/current_branch1/cime/src/drivers/nuopc/mediator/med_phases_aofluxes_mod.F90(500): error #8284: If the actual argument is scalar, the dummy argument shall be scalar unless the actual argument is of type character or is an element of an array that is not assumed shape, pointer, or polymorphic.   [DUU10N]
    call shr_flux_atmocn (&
---------^
compilation aborted for /glade/work/sacks/ctsm_code/current_branch1/cime/src/drivers/nuopc/mediator/med_phases_aofluxes_mod.F90 (code 1)
gmake: *** [med_phases_aofluxes_mod.o] Error 1
```

My best guess is that the fairly old cime version being used in CTSM is incompatible with the nuopc version being used there.

So then I tried checking out cime master, but got a different error at create_newcase time:

```
$ ./create_test --no-run SMS_D_Vnuopc.f10_f10_musgs.I2000Clm50SpRsGs.cheyenne_intel
Testnames: ['SMS_D_Vnuopc.f10_f10_musgs.I2000Clm50SpRsGs.cheyenne_intel']
Using project from env PROJECT: P93300606
Creating test directory /glade/scratch/sacks/SMS_D_Vnuopc.f10_f10_musgs.I2000Clm50SpRsGs.cheyenne_intel.20191120_135837_xoyf1g
RUNNING TESTS:
  SMS_D_Vnuopc.f10_f10_musgs.I2000Clm50SpRsGs.cheyenne_intel
Starting CREATE_NEWCASE for test SMS_D_Vnuopc.f10_f10_musgs.I2000Clm50SpRsGs.cheyenne_intel with 1 procs
Finished CREATE_NEWCASE for test SMS_D_Vnuopc.f10_f10_musgs.I2000Clm50SpRsGs.cheyenne_intel in 1.270961 seconds (FAIL). [COMPLETED 1 of 1]
    Case dir: /glade/scratch/sacks/SMS_D_Vnuopc.f10_f10_musgs.I2000Clm50SpRsGs.cheyenne_intel.20191120_135837_xoyf1g
    Errors were:
        ERROR: Looks like job groups have already been created

At test-scheduler close, state is:
FAIL SMS_D_Vnuopc.f10_f10_musgs.I2000Clm50SpRsGs.cheyenne_intel (phase CREATE_NEWCASE)
    Case dir: /glade/scratch/sacks/SMS_D_Vnuopc.f10_f10_musgs.I2000Clm50SpRsGs.cheyenne_intel.20191120_135837_xoyf1g
test-scheduler took 1.40396308899 seconds
```

**@mvertens my plan is to bring this to master as is for now. I'm inclined to skip even adding the apparently-needed `externals = ../Externals_cime.cfg` line, since that might give the impression that a nuopc build works here, which it apparently doesn't. Let me know if you object with this plan.**

**Comment by mvertens** [comment]
@billsacks - the problem with the atmocn flux problem is that cmeps is
using the updated cime version for the atm/ocn flux calculation where the
arguments have been changed by an e3sm update. So if you are using an older
version of cime - you cannot just bring the latest cmeps version in. Lets
discuss as soon as possible.

**Comment by billsacks** [comment]
Working with @mvertens - to get past the last problem I had (`Looks like job groups have already been created`), I needed to update cmeps to the latest (181ff1e).

**Pull request merged**